omisego-sdk/src/main/java/co/omisego/omisego/websocket/SocketCallback.kt

```diff
@@ -0,0 +1,84 @@
+package co.omisego.omisego.websocket
+
+/*
+ * OmiseGO
+ *
+ * Created by Phuchit Sirimongkolsathien on 8/5/2018 AD.
+ * Copyright © 2017-2018 OmiseGO. All rights reserved.
+ */
+
+import co.omisego.omisego.model.APIError
+import co.omisego.omisego.model.socket.SocketReceive
+import co.omisego.omisego.model.transaction.consumption.TransactionConsumption
+
+/**
+ * A web socket connection callback that executed when the web socket client is connected to the server or disconnected from the server.
+ */
+interface SocketConnectionCallback {
+    /**
+     * Invoked when the web socket client has connected to the eWallet web socket API successfully.
+     */
+    fun onConnected()
+
+    /**
+     * Invoked when the web socket client has disconnected from the eWallet web socket API.
+     *
+     * @param throwable (Optional) The exception might be raised if the web socket was not disconnected successfully.
+     */
+    fun onDisconnected(throwable: Throwable?)
+}
+
+interface SocketChannelCallback {
+    /**
+     * Invoked when the client have been joined the channel successfully.
+     *
+     * @param topic A topic indicating which channel will be joined.
+     */
+    fun onJoinedChannel(topic: String)
+
+    /**
+     * Invoked when the client have been left the channel successfully.
+     *
+     * @param topic A topic indicating which channel will be joined.
+     */
+    fun onLeftChannel(topic: String)
+
+    /**
+     * Invoked when something goes wrong while connecting to a channel.
+     *
+     * @param apiError An [APIError] instance for explaining the failure reason.
+     */
+    fun onError(apiError: APIError)
+}
+
+sealed class SocketCustomEventCallback {
+    /**
+     * A callback for every event in the SocketCustomEvent (see SocketEvent.kt).
+     */
+    abstract class AnyEventCallback : AnySocketEvent, SocketCustomEventCallback()
+
+    /**
+     * A callback for the [SocketTransactionRequestEvent]
+     */
+    abstract class TransactionRequestCallback : SocketTransactionRequestEvent, SocketCustomEventCallback()
+
+    /**
+     * A callback for the [SocketTransactionConsumptionEvent]
+     */
+    abstract class TransactionConsumptionCallback : SocketTransactionConsumptionEvent, SocketCustomEventCallback()
+}
+
+private interface SocketTransactionRequestEvent {
```

Comment:
Having an interface like this named `Event` feels weird.
You wouldn't except an **event** object to have multiple "callback" methods to implement 🤔 
Are the interface really needed? Why not "inline" directly inside the declaration of the `callback` class. 
Would feel more natural to have a `callback` class declare **callback** methods

Reply:
Or maybe it was to keep a "clean" and short declaration of the sealed class?
If so, not sure it's worth to "complicate" the code, it'll be in the same file anyway.
At least if it's in the class, you can fold the class block in the IDE and it's easier to understand